### PR TITLE
Impact cleanup/review

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

First pass through Impact class before UI work. UI migrations will be a separate PR after this.

### CODE REVIEW

**Review by AdrienClairembault — APPROVED** [review]

File: src/Impact.php [hunk 1]
```diff
@@ -744,6 +738,10 @@ public static function bfs(array $graph, array $a, array $b, int $direction)
                 $queue[] = $nextNode;
             }
         }
+
+        // No path found
+        //TODO Ask if this should throw an exception instead
```
Comment (AdrienClairembault) [inline src/Impact.php:743]: I agree for an exception (in another PR)

**Comment by trasher** [comment]
CS is failing

**Review by AdrienClairembault — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]
Please add an entry in changelog when you delete a public method. Some plugins may use it, and it will be easier for plugins developpers to have a complete list of deleted methods.

I have done it for the current PR. Also, I added a generic `Many methods signatures have been updated to specify both their return type and the types of their parameters.` entry in changelog.

**Comment by AdrienClairembault** [comment]
Wouldn't you need to also list changes to protected methods on a non final classes ?

**Comment by cedric-anne** [comment]
> Wouldn't you need to also list deletion of protected methods on non-final classes ?

Yes, also.

**Pull request merged**